src/core/components/AuthenticateButton/index.js

```diff
@@ -53,9 +55,17 @@ export const mapStateToProps = (state) => ({
 });
 
 export const mapDispatchToProps = (dispatch) => ({
-  handleLogOut() {
+  handleLogOut({ api }) {
+    const { location } = window;
+    if (config.get('apiHost') === `${location.protocol}//${location.host}`) {
+      // Deployed config, server manages cookie.
+      return logOutFromServer({ api })
+        .then(() => dispatch(logOutUser()));
```

Comment:
Yeah, I tried that in https://github.com/mozilla/addons-frontend/pull/1496 but couldn't think of a way to clear the error. I like the idea of adding a 'close' button so that the user can clear it. I'm not sure yet if we have a stable hook for when a new page gets rendered but maybe we do.